instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/FieldMapper.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import io.opentelemetry.api.trace.Span;
+import java.util.function.Function;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.SdkResponse;
+
+class FieldMapper {
+
+  private final Serializer serializer;
+  private final MethodHandleFactory methodHandleFactory;
+
+  FieldMapper() {
+    serializer = new Serializer();
+    methodHandleFactory = new MethodHandleFactory();
+  }
+
+  FieldMapper(Serializer serializer, MethodHandleFactory methodHandleFactory) {
+    this.methodHandleFactory = methodHandleFactory;
+    this.serializer = serializer;
+  }
+
+  void mapFields(SdkRequest sdkRequest, AwsSdkRequest request, Span span) {
+    mapFields(
+        field -> sdkRequest.getValueForField(field, Object.class).orElse(null),
+        FieldMapping.Type.REQUEST,
+        request,
+        span);
+  }
+
+  void mapFields(SdkResponse sdkResponse, AwsSdkRequest request, Span span) {
+    mapFields(
+        field -> sdkResponse.getValueForField(field, Object.class).orElse(null),
+        FieldMapping.Type.RESPONSE,
+        request,
+        span);
+  }
+
+  private void mapFields(
+      Function<String, Object> fieldValueProvider,
+      FieldMapping.Type type,
+      AwsSdkRequest request,
+      Span span) {
+    for (FieldMapping fieldMapping : request.fields(type)) {
+      mapFields(fieldValueProvider, fieldMapping, span);
+    }
+    for (FieldMapping fieldMapping : request.type().fields(type)) {
+      mapFields(fieldValueProvider, fieldMapping, span);
+    }
+  }
+
+  private void mapFields(
+      Function<String, Object> fieldValueProvider, FieldMapping fieldMapping, Span span) {
+    // traverse path
+    String[] path = fieldMapping.getFields();
+    Object target = fieldValueProvider.apply(camelCase(path[0]));
```

Comment:
This `camelCase()` call wouldn't be needed if fields in `AwsSdkRequestType` were already camelCased, would it?

Reply:
So, here's the deal - in AWS model, fields are CamelCased, but getters start with lowercase (FieldName vs fieldName). First field in a path is retrieved by getFieldValue call which takes field name, rest is retrieved with getter call. So either way we're going to transform the name, assuming we want to stay consistent with naming (avoid putting first camel cased and rest lowercased).